packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -146,13 +148,19 @@ class DraggableScrollableSheet extends StatefulWidget {
   /// The default value is true.
   final bool expand;
 
+  /// A list of target points that the widget should snap to. [minChildSize] and
+  /// [maxChildSize] are implicitly snapped to and should not be specified here-
+  /// pass an empty list to snap the widget between min and max.
+  final List<double>? snapTargets;
```

Comment:
This would be a natural place to use a set (order doesn't matter, duplicates don't make any conceptual sense). However, I hope to next tackle: #45009.

As a part of that PR I'd like to create a Future<void> animateToSnapPoint(int index) function where users can programmatically animate to a given snap point based on it's index in snapPoints. This functionality necessitates a list.

Additionally, the current implementation assumes that you only ever want to snap to the closest target. We could let users specify snapping thresholds which would probably be an additionally list (or map?) and would required that `snapTargets` be both a list and sorted.

Reply:
The behavior/expectations should be documented.

Reply:
It also seems strange to me that an empty list automatically does anything. Do we always snap between min and max if values are otherwise specified?

Reply:
As is I haven't put any restrictions on `snapTargets`. If we later decide to add something like `animateToSnapPoint` or `snapThresholds`, we'd have to assert that `snapTargets` is an ordered list without duplicates. I'm wondering if it'd be better to add this assert in now to avoid a potential breaking change down the line?

We always snap between min and max. Under the hood I'm just adding min and max to the front and end of the user specified list, respectively. I agree that it's a little weird. Technically, you can specify min and max in `snapTargets`, it just won't have any effect. This mirrors the behavior of `Gradients.stops`.

Reply:
> It also seems strange to me that an empty list automatically does anything.

+1 to that.

Reply:
If snapping is enabled, it's not very coherent to NOT treat min and max as snapping targets-what would it mean to have parts of the draggable range snap and other parts not?

Right now I'm using a non-null `snapTargets` to indicate that snapping is desired. How about I:
- Introduce another argument `Bool snap` that enables or disables snapping
- Retain current behavior where `snapTargets` always implicitly includes min and max whether or not the user specifies them (consistent with https://api.flutter.dev/flutter/painting/Gradient/stops.html)
- Make `snapTargets` a non-nullable list with the empty list as the default value

This way the user never needs to provide an empty list to trigger snapping behavior.

Reply:
Implemented the above solution.